test/integration/clusters/custom_static_cluster.cc

```diff
@@ -0,0 +1,88 @@
+#include <chrono>
+#include <list>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/http/codec.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/network/address_impl.h"
+#include "common/upstream/cluster_factory_impl.h"
+
+#include "server/transport_socket_config_impl.h"
+
+#include "test/common/upstream/utility.h"
+#include "test/integration/clusters/cluster_factory_config.pb.h"
+#include "test/integration/clusters/cluster_factory_config.pb.validate.h"
+#include "test/test_common/registry.h"
+
+namespace Envoy {
+
+class CustomStaticCluster : public Upstream::ClusterImplBase {
```

Comment:
This is basically the same code you use in your unit test, right? Is it possible to just share it between the two tests?